sdk/core/azure-core/src/main/java/com/azure/core/models/HttpResponseInnerError.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models;
+
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The inner error of a {@link HttpResponseError}.
+ */
+public final class HttpResponseInnerError {
+
+    @JsonProperty(value = "code")
+    private String code;
+
+    @JsonProperty(value = "innererror")
+    private HttpResponseInnerError innerError;
```

Comment:
I remember working with services that had inner errors that included a `message`, will that not the case for this generic type?

Reply:
The [REST guideline](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#errorresponse--object) for this type does not include `message` in the inner error type. If required, the inner error can be extended by individual services to include `message`.